apollo-router/src/services/router/service.rs

```diff
@@ -812,12 +818,47 @@ impl RouterService {
         Ok(graphql_requests)
     }
 
-    fn count_errors(errors: &[graphql::Error]) {
+    fn count_errors(&self, errors: &[graphql::Error], context: &Context) {
+        let codes = errors
+            .iter()
+            .map(|e| e.extensions.get("code").and_then(|c| c.as_str()))
```

Comment:
IIUC for errors without a code, it would count against the "empty" code e.g. `code=""`.  I think we could consider defining an explicit `UNKNOWN` code but it seems fine to support this with an empty code.

Reply:
Yeah this is an empty string. I can easily add a fallback code but I think an empty string is just as good.